ArduCopter/Copter.cpp

```diff
@@ -634,6 +634,12 @@ void Copter::loop_rate_logging()
     if (should_log(MASK_LOG_FTN_FAST) && !using_rate_thread) {
         AP::ins().write_notch_log_messages();
     }
+#endif
+#if HAL_WITH_ESC_TELEM
+    // associate full rate ESC data with full rate notch data
+    if (should_log(MASK_LOG_MOTBATT) && should_log(MASK_LOG_FTN_FAST)) {
```

Comment:
Allocate a new logging bit.

Pass an an interval into the library, let it to the work.

My preference would be for a logging interval as a parameter in the library and leave the vehicles out of it entirely.